### PR TITLE
Add a step for remapping iceberg climatologies 

### PR DESCRIPTION
**Pull request by xylar**
This is used in E3SM runs with data iceberg freshwater fluxes.

Checklist
* [x] Developer's Guide has been updated
* [x] API documentation in the Developer's Guide (`api.rst`) has any new or modified class, method and/or functions listed
* [x] Documentation has been [built locally](https://mpas-dev.github.io/compass/latest/developers_guide/building_docs.html) and changes look as expected
* [x] Document (in a comment titled `Testing` in this PR) any testing that was used to verify the changes

### CODE REVIEW

File: docs/developers_guide/ocean/test_groups/global_ocean.rst [hunk 1]
```diff
@@ -1352,6 +1352,12 @@ The test case is made up of 10 steps:
     Hydrographic Climatology (PHC) to the MPAS mesh.  This dataset is used in
     E3SM for compsets with data atmospheric/land forcing.
 
+:py:class:`compass.ocean.tests.global_ocean.files_for_e3sm.remap_iceberg_climatology.RemapIcebergClimatology`
+    is used to remap freshwater fluxes from an iceberg climatology from
+    `Merino et al. (2016) <https://doi.org/10.1016/j.ocemod.2016.05.001>`_ to
```
Comment (xylar) [inline docs/developers_guide/ocean/test_groups/global_ocean.rst:1357]: @darincomeau, can you verify that this is the right reference for the iceberg climatology?

**Comment by xylar** [comment]
I'm seeing a seam at +/- 180 so I need to figure out what's going wrong there.  Probably the non-monotonic longitude.

output_with_seam

File: docs/developers_guide/ocean/test_groups/global_ocean.rst (hunk 1 above)
Comment (darincomeau) [reply docs/developers_guide/ocean/test_groups/global_ocean.rst:1357]: confirmed!

Comment (xylar) [reply docs/developers_guide/ocean/test_groups/global_ocean.rst:1357]: Thanks!

**Comment by xylar** [comment]
Okay, finally got rid of the seam:
output_no_seam

**Comment by jonbob** [comment]
I remember the same issue when Darin and I were first working on this....

**Comment by xylar** [comment]
There really is a block of zeros on the east and west sides of the data!

**Comment by xylar** [comment]
@jonbob and @darincomeau, I'm not necessarily asking for a review of the code here (though that's fine if you want to).  But it would be nice to have you double check that the 4 files:
```
/lcrc/group/e3sm/data/inputdata/ice/mpas-seaice/ECwISC30to60E3r1/Iceberg_Climatology_Merino.ECwISC30to60E3r1.20230815.nc
/lcrc/group/e3sm/data/inputdata/ice/mpas-seaice/IcoswISC30E3r1/Iceberg_Climatology_Merino.IcoswISC30E3r1.20230815.nc
/lcrc/group/e3sm/data/inputdata/ice/mpas-seaice/RRSwISC6to18E3r1/Iceberg_Climatology_Merino.RRSwISC6to18E3r1.20230818.nc
/lcrc/group/e3sm/data/inputdata/ice/mpas-seaice/SOwISC12to60E3r1/Iceberg_Climatology_Merino.SOwISC12to60E3r1.20230815.nc
```
look okay by whatever approach you would like to take to look at them.  I looked at January of `Iceberg_Climatology_Merino.IcoswISC30E3r1.20230815.nc` in paraview but that's the most I've done so far.

**Comment by xylar** [comment]
## Testing

I ran the `files_for_e3sm/remap_iceberg_climatology` step designed for existing meshes for the 4 meshes listed above.  I also ran it for the `Icos240E3r1` within the full compass mesh generation workflow.  All test cases succeeded in producing the desired output file, and the `Icos240E3r1` and `IcoswISC30E3r1` January outputs looked as expected in Paraview.

**Comment by xylar** [comment]
@jonbob and @darincomeau, unless you have feedback on the iceberg melt files, I think I'll merge this based on my testing.

**Pull request merged**